Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,262 @@
+            {
+                return InValidOutput;
+            }
+
+            var richestPerson = new StringBuilder();
+            var highestBalance = FindRichest(peopleAndBalances, richestPerson);
+
+            var persons = richestPerson.ToString();
+
+            if (persons.Contains(','))
+            {
+                return $"{persons} are the richest people. ¤{highestBalance}.";
+            }
+            else
+            {
+                return $"{persons} is the richest person. ¤{highestBalance}.";
+            }
+        }
+
+        private static decimal FindRichest(string[] peopleAndBalances, StringBuilder RichestPerson)
+        {
+            var HighestBalance = decimal.MinValue;
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var highestAmountOfPerson = decimal.Parse(currentPersonData[currentPersonData.Length - 1]);
+
+                if (highestAmountOfPerson == HighestBalance)
+                {
+                    if (i == peopleAndBalances.Length - 1)
+                    {
+                        RichestPerson.Append(" and ");
+                    }
+                    else
+                    {
+                        RichestPerson.Append(", ");
+                    }
+
+                    RichestPerson.Append(currentPersonData[0]);
+                }
+
+                if (highestAmountOfPerson > HighestBalance)
+                {
+                    HighestBalance = highestAmountOfPerson;
+                    var currentOne = RichestPerson.ToString();
+
+                    if (string.IsNullOrEmpty(currentOne))
+                    {
+                        RichestPerson.Append(currentPersonData[0]);
+                    }
+                    else
+                    {
+                        RichestPerson.Clear();
+                        RichestPerson.Append(currentPersonData[0]);
+                    }
+                }
+            }
+
+            return HighestBalance;
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var poorestPerson = new StringBuilder();
+            var lowestBalance = FindPoorest(peopleAndBalances, poorestPerson);
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+            NumberFormatInfo noParens = (NumberFormatInfo)CultureInfo.CurrentCulture.NumberFormat.Clone();
+            noParens.CurrencyNegativePattern = 1;
+            var lowestBalanceString = lowestBalance.ToString("C0", noParens);
+
+            var persons = poorestPerson.ToString();
+
+            if (persons.Contains(','))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4937470)